src/EFCore.PG/Extensions/NpgsqlIndexBuilderExtensions.cs

```diff
@@ -554,6 +554,88 @@ public static class NpgsqlIndexBuilderExtensions
+        /// <returns>A builder to further configure the index.</returns>
+        public static IndexBuilder<TEntity> IsCreatedConcurrently<TEntity>([NotNull] this IndexBuilder<TEntity> indexBuilder, bool createdConcurrently)
+            => (IndexBuilder<TEntity>)IsCreatedConcurrently((IndexBuilder)indexBuilder, createdConcurrently);
+
+        /// <summary>
+        /// When this option is used, PostgreSQL will build the index without taking any locks that prevent concurrent inserts,
+        /// updates, or deletes on the table; whereas a standard index build locks out writes (but not reads) on the table until it's done.
+        /// </summary>
+        /// <remarks>
+        /// https://www.postgresql.org/docs/current/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY
+        /// </remarks>
+        /// <param name="indexBuilder">The builder for the index being configured.</param>
+        /// <param name="createdConcurrently">A value indicating whether the index is created with the "concurrently" option.</param>
+        /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+        /// <returns>A builder to further configure the index.</returns>
+        public static IConventionIndexBuilder IsCreatedConcurrently(
+            [NotNull] this IConventionIndexBuilder indexBuilder,
+            bool? createdConcurrently,
+            bool fromDataAnnotation = false)
+        {
+            if (indexBuilder.CanSetIsCreatedConcurrently(createdConcurrently, fromDataAnnotation))
+            {
+                indexBuilder.Metadata.SetIsCreatedConcurrently(createdConcurrently);
+
+                return indexBuilder;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Returns a value indicating whether concurrent creation for the index can be set.
+        /// </summary>
+        /// <remarks>
+        /// https://www.postgresql.org/docs/current/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY
+        /// </remarks>
+        /// <param name="indexBuilder">The builder for the index being configured.</param>
+        /// <param name="createdConcurrently">A value indicating whether the index is created with the "concurrently" option.</param>
+        /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+        /// <returns>A builder to further configure the index.</returns>
+        public static bool CanSetIsCreatedConcurrently(
+            [NotNull] this IConventionIndexBuilder indexBuilder,
+            bool? createdConcurrently,
+            bool fromDataAnnotation = false)
+        {
+            Check.NotNull(indexBuilder, nameof(indexBuilder));
+
+            return indexBuilder.CanSetAnnotation(NpgsqlAnnotationNames.CreatedConcurrently, createdConcurrently, fromDataAnnotation);
```

Comment:
A good candidate for one lining.

Reply:
Not a bad idea but can be done for other properties too, so leaving out of scope for now. It may also be good to leave this way to remain close to the EF Core style.